src/lib.rs

```diff
@@ -228,6 +232,19 @@ impl RustNotify {
         Ok(py_changes)
     }
 
+    /// https://github.com/PyO3/pyo3/issues/1205#issuecomment-1164096251 for advice on `__enter__`
+    pub fn __enter__(slf: Py<Self>) -> Py<Self> {
+        slf
+    }
+
+    pub fn close(&mut self) {
+        self.watcher = WatcherEnum::None;
```

Comment:
Can you override `__del__` to issue a ResourceWarning is the watcher is not None?

Reply:
Maybe, but I'm not sure it's worth it - if you actually delete the instance the thread will be closed. So technically `__del__` is just as correct as `__exit__` or `close`.

Also `RustNotify` is not supposed to be used directly and both the public methods which use take care of clean up.

I think what we have here is sufficient.

Reply:
> - if you actually delete the instance the thread will be closed. 

Yes this is the purpose of ResourceWarning in a `__del__` method

Reply:
pytest hooks into this GC/ResourceWarning system to fail the test suite if an explicit close is forgotten

Reply:
But omitting to call close is fine in general.

If you'd like to submit a PR, feel free.

Reply:
Apart from your warning suggestion on `__del__`, can you confirm if this works?

Reply:
On reflection I think you're right that we should raise a resource warning, fuller answer below.

Reply:
I've created #166 to keep track of adding a resource warning. I think htis can be  merged through since we shouldn't wait for pyo3 to resolve the original bug here.